In [88]:
import os

import pandas as pd

In [89]:

in_relative_path = "../exports_covoit/"
in_absolute_path = os.path.abspath(os.path.join(os.getcwd(), in_relative_path))
print(in_absolute_path)

/home/mattcoolioh/voyages_covoit_fr/exports_covoit


In [90]:
os.listdir(in_absolute_path)

['2023-10.csv',
 '2023-05.csv',
 '2023-07.csv',
 '2024-03.csv',
 '2024-01.csv',
 '2023-06.csv',
 '2022-06.csv',
 '2022-08.csv',
 '2023-02.csv',
 '2023-12.csv',
 '2024-07.csv',
 '2024-05.csv',
 '2022-07.csv',
 '2024-04.csv',
 '2022-11.csv',
 '2023-08.csv',
 '2023-01.csv',
 '2022-12.csv',
 '2024-02.csv',
 '2023-09.csv',
 '2023-04.csv',
 '2023-11.csv',
 '2022-10.csv',
 '2023-03.csv',
 '2022-09.csv',
 '2024-06.csv']

In [91]:
df = pd.read_csv(f"{in_absolute_path}/2023-06.csv", sep=";", engine="python")
df.head()

journey_id                               trip_id  \
0    15797768  fd2ba71d-2641-47f6-bc10-23d85cd6b481   
1    15901731  fa8b2297-e5e9-4112-b4e2-8905973b9b45   
2    15901726  997e3c6e-0215-4d05-9b94-8c28b9d5ad7a   
3    15833705  f78748ec-b651-4b5f-bc55-fb59d7e0a906   
4    15797699  c482a7ea-c89b-440b-89df-7361d5c90ccb   

      journey_start_datetime journey_start_date journey_start_time  \
0  2023-06-01T00:00:00+02:00         2023-06-01           00:00:00   
1  2023-06-01T00:00:00+02:00         2023-06-01           00:00:00   
2  2023-06-01T00:00:00+02:00         2023-06-01           00:00:00   
3  2023-06-01T00:00:00+02:00         2023-06-01           00:00:00   
4  2023-06-01T00:00:00+02:00         2023-06-01           00:00:00   

   journey_start_lon  journey_start_lat journey_start_insee  \
0              2.446             48.844               75112   
1              1.973             48.790               78208   
2              2.165             48.803               78686   
3              2.281             48.888               92051   
4              3.973             43.542               34154   

  journey_start_department journey_start_town  ... journey_end_insee  \
0                       75              Paris  ...             92025   
1                       78          Élancourt  ...             78165   
2                       78           Viroflay  ...             92062   
3                       92  Neuilly-sur-Seine  ...             75114   
4                       34            Mauguio  ...             34172   

  journey_end_department      journey_end_town  \
0                     92              Colombes   
1                     78  Les Clayes-sous-Bois   
2                     92               Puteaux   
3                     75                 Paris   
4                     34           Montpellier   

                journey_end_towngroup journey_end_country  passenger_seats  \
0            Métropole du Grand Paris              France                1   
1     CA de Saint-Quentin-en-Yvelines              France                1   
2            Métropole du Grand Paris              France                1   
3            Métropole du Grand Paris              France                1   
4  Montpellier Méditerranée Métropole              France                1   

   operator_class journey_distance journey_duration has_incentive  
0               C            27976               42           OUI  
1               C             7430                9           OUI  
2               C            15891               18           OUI  
3               C             8117               28           OUI  
4               C            26392               40           OUI  

[5 rows x 27 columns]

In [92]:
df.dtypes

journey_id                    int64
trip_id                      object
journey_start_datetime       object
journey_start_date           object
journey_start_time           object
journey_start_lon           float64
journey_start_lat           float64
journey_start_insee          object
journey_start_department     object
journey_start_town           object
journey_start_towngroup      object
journey_start_country        object
journey_end_datetime         object
journey_end_date             object
journey_end_time             object
journey_end_lon             float64
journey_end_lat             float64
journey_end_insee            object
journey_end_department       object
journey_end_town             object
journey_end_towngroup        object
journey_end_country          object
passenger_seats               int64
operator_class               object
journey_distance              int64
journey_duration              int64
has_incentive                object
dtype: object

Chaque ligne correspond à un trajet de covoiturage, c"est à dire un couple passager / conducteur. A chaque passager est donc affecté un trajet.

Exemple : un conducteur réalise un déplacement avec deux passagers différents au sein de son véhicule, le nombre de trajets réalisés et de 2. Ceci se traduit par deux lignes.

Colonnes à garder :  
- journey_id  
Identifiant unique affecté à un couple passager/conducteur.  
- trip_id  
Identifiant permettant de regrouper plusieurs couples passager/conducteur dans un même véhicule.  
Exemple : c4124bb1-d8a4-487c-b4d9-367b931ee8ce  
- journey_start_datetime  
Exemple : 2019-10-31T23:00:00.000Z  
Date et heure du départ au format ISO 8601 (YYYY-MM-DDThh:mm:ssZ).  
L"heure est exprimée en UTC (Coordinated Universal Time). UTC n"est pas ajusté sur l"heure d"été et hiver !  
- journey_start_insee  
Exemple : 91377  
Code INSEE commune ou arrondissement du point de départ (prise en charge passager). Déterminé via les coordonnées géographiques  
- journey_start_town  
Exemple : Massy  
Commune du point de départ (prise en charge passager). Déterminé via les coordonnées géographiques  
- journey_end_insee  
Exemple : 91122  
Code INSEE commune ou arrondissement du point d"arrivée (dépose passager).Déterminé via les coordonnées géographiques  
- journey_end_town  
Exemple : Bures-sur-Yvette  
Commune du point d"arrivée (dépose passager). Déterminé via les coordonnées géographiques  
- passenger_seats  
Exemple : 2  
Nombre de sièges réservés par l"occupant passager  
- journey_distance  
Exemple: 34523  
Distance parcourue en mètres. Information envoyée par l"opérateur  
- journey_duration  
Exemple: 60  
Durée du trajet en minutes. Information envoyée par l"opérateur  

Colonnes à supprimer : 
- journey_start_date
Exemple : 2019-10-31
Date départ au format YYYY-MM-DD
- journey_start_time
Exemple : 02:20:001
Heure de départ au format hh:mm:ss
- journey_start_lat
Exemple : 48.725
Latitude du point de départ (prise en charge passager) comprise entre 90deg et -90deg décimaux en datum WSG-84
- journey_start_lon
Exemple : 2.261
Longitude du point de départ (prise en charge passager) comprise entre 180deg et -180deg décimaux en datum WSG-84
- journey_start_postcode Supprimé à partir de Juin 2022
Exemple : 91300
Code postal du point de départ (prise en charge passager). Déterminé via les coordonnées géographiques
- journey_start_towngroup
Exemple : Ile-De-France Mobilites
EPCI ou AOM de départ. Déterminé via les coordonnées géographiques. Peut être vide
- journey_start_department
Exemple : 38
Département du point de départ (prise en charge passager). Déterminé via les coordonnées géographiques
- journey_start_country
Exemple : France
Pays du point de départ (prise en charge passager). Déterminé via les coordonnées géographiques
- journey_end_datetime
Exemple : 2019-10-31T23:15:00.000Z
Date et heure de l"arrivée au format ISO 8601 (YYYY-MM-DDThh:mm:ssZ).
L"heure est exprimée en UTC (Coordinated Universal Time). UTC n"est pas ajusté sur l"heure d"été et hiver !
- journey_end_date
Exemple : 2019-10-31
Date d"arrivée au format YYYY-MM-DD
- journey_end_time
Exemple : 02:20:001
Heure d"arrivéet au format hh:mm:ss
- journey_end_lat
Exemple : 48.695
Latitude du point d"arrivée (dépose passager) comprise entre 90deg et -90deg décimaux en datum WSG-84.
- journey_end_lon
Exemple : 2.162
Longitude du point d"arrivée (dépose passager) comprise entre 180deg et -180deg décimaux en datum WSG-84
- journey_end_postcode Supprimé à partir de Juin 2022
Exemple : 91440
Code postal du point d"arrivée (dépose passager). Déterminé via les coordonnées géographiques
- journey_end_department
Exemple : 38
Département du point d"arrivée (dépose passager). Déterminé via les coordonnées géographiques
- journey_end_towngroup
Exemple : Ile-De-France Mobilites
EPCI ou AOM d"arrivée. Déterminé via les coordonnées géographiques. Peut être vide
- journey_end_country
Exemple : France
Pays du point d"arrivée (dépose passager)
- operator_class
Exemple: C
Classe de preuve de covoiturage
- has_incentive
Exemple: OUI
Si le trajet est incité ou non par un opérateur ou une campagne paramétrée sur le registre. Attention : à date cette donnée est renseignée partiellement et ne peut donner lieu à des analyses et conclusions.

In [93]:
colonne_a_garder = [
    "journey_id",
    "trip_id",
    "journey_start_date",
    "journey_start_insee",
    "journey_start_town",
    "journey_end_insee",
    "journey_end_town",
    "passenger_seats",
    "journey_distance",
    "journey_duration"
]
df = df[colonne_a_garder]
df.sample(5)

journey_id                               trip_id journey_start_date  \
108537    16035715  9c5321b1-0343-4354-81a1-7383fc20def4         2023-06-05   
558444    16438626  680a9c91-2be3-4a4e-ae79-24b4e0dab8bd         2023-06-21   
220949    16120399  5454e493-63a1-4f45-8b3c-072b3670f920         2023-06-08   
142934    15950118  16d3da7c-09f4-46c1-8dcd-79540a488c74         2023-06-06   
650369    16510210  e5d5949b-6e14-4c61-ac5c-8ada5b0e9d7b         2023-06-26   

       journey_start_insee journey_start_town journey_end_insee  \
108537               91692           Les Ulis             91312   
558444               85191   La Roche-sur-Yon             44043   
220949               78646         Versailles             78092   
142934               76217             Dieppe             76245   
650369               62386             Grenay             62801   

       journey_end_town  passenger_seats  journey_distance  journey_duration  
108537             Igny                1             12080                 9  
558444          Clisson                1             55035                45  
220949         Bougival                1              7986                11  
142934         Eslettes                1             47748                38  
650369          Souchez                1              8869                16

In [94]:
df.describe()

journey_id  passenger_seats  journey_distance  journey_duration
count  7.840680e+05    784068.000000      7.840680e+05     784068.000000
mean   1.624432e+07         1.000509      2.441856e+04         25.338422
std    2.382115e+05         0.030696      1.820413e+04         13.909284
min    1.508868e+07         1.000000      0.000000e+00          0.000000
25%    1.604213e+07         1.000000      1.133400e+04         15.000000
50%    1.624311e+07         1.000000      2.105250e+04         24.000000
75%    1.644570e+07         1.000000      3.265225e+04         33.000000
max    1.672015e+07         4.000000      1.129662e+06       1424.000000

In [95]:
df.describe(include="object")

trip_id journey_start_date  \
count                                 784068             784068   
unique                                652692                 30   
top     8abbc98e-e56b-4b2a-a7a6-b9c142f91c46         2023-06-06   
freq                                      16              36852   

       journey_start_insee journey_start_town journey_end_insee  \
count               784068             758229            784068   
unique                6544               6407              6372   
top                  34172        Montpellier             34172   
freq                 12827              12827             12911   

       journey_end_town  
count            756630  
unique             6242  
top         Montpellier  
freq              12911

In [96]:
df["passenger_seats"].value_counts()

passenger_seats
1    783805
2       161
3        68
4        34
Name: count, dtype: int64

Il y a une très grande majoritée de réservation de covoiturage avec un seul passager.

In [71]:
df[df["journey_start_town"].isna()]["journey_start_insee"].value_counts()

journey_start_insee
99140    10484
99138     6812
99137     4602
99127     2308
99131     1323
99109      298
99134       12
Name: count, dtype: int64

Les quelques données manquantes en "journey_start_town" et "journey_end_town" correspondent à trajets avec des villes dans un pays étranger.  
Majoritairement Monaco (Code INSEE 99138), la Suisse (Code INSEE 99140) et l"Italie (Code INSEE 99127) pour juin 2022.

In [74]:
df.groupby(["journey_start_date"])["passenger_seats"].sum().reset_index()


journey_start_date  passenger_seats
0          2023-06-01            35950
1          2023-06-02            29306
2          2023-06-03            11770
3          2023-06-04            10213
4          2023-06-05            34567
5          2023-06-06            36874
6          2023-06-07            33115
7          2023-06-08            35082
8          2023-06-09            28090
9          2023-06-10            11213
10         2023-06-11             9218
11         2023-06-12            33482
12         2023-06-13            36525
13         2023-06-14            32454
14         2023-06-15            33939
15         2023-06-16            27268
16         2023-06-17            11072
17         2023-06-18             9584
18         2023-06-19            32560
19         2023-06-20            35581
20         2023-06-21            31341
21         2023-06-22            32090
22         2023-06-23            25253
23         2023-06-24            10371
24         2023-06-25             8822
25         2023-06-26            31086
26         2023-06-27            33405
27         2023-06-28            29144
28         2023-06-29            30964
29         2023-06-30            24128

In [87]:
import numpy as np
df["journey_distance_tot_passengers"] = (df["journey_distance"] * df["passenger_seats"])
df.groupby("journey_start_date").agg(
    nb_passagers=("passenger_seats", np.sum),
    nb_trajets=("trip_id", "nunique"),
    total_distance=("journey_distance_tot_passengers", np.sum)).reset_index()

/tmp/ipykernel_125150/3101316476.py:3: FutureWarning: The provided callable <function sum at 0x7facbc34b250> is currently using SeriesGroupBy.sum. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "sum" instead.
  df.groupby("journey_start_date").agg(
/tmp/ipykernel_125150/3101316476.py:3: FutureWarning: The provided callable <function sum at 0x7facbc34b250> is currently using SeriesGroupBy.sum. In a future version of pandas, the provided callable will be used directly. To keep current behavior pass the string "sum" instead.
  df.groupby("journey_start_date").agg(


journey_start_date  nb_passagers  nb_trajets  total_distance
0          2023-06-01         35950       29952       879496769
1          2023-06-02         29306       24579       706799866
2          2023-06-03         11770        9703       256842065
3          2023-06-04         10213        8263       237913757
4          2023-06-05         34567       28622       858946125
5          2023-06-06         36874       30394       915769307
6          2023-06-07         33115       27399       811846280
7          2023-06-08         35082       29027       857893189
8          2023-06-09         28090       23565       669620195
9          2023-06-10         11213        9301       234915478
10         2023-06-11          9218        7538       213686752
11         2023-06-12         33482       27759       827018632
12         2023-06-13         36525       30081       897638919
13         2023-06-14         32454       26675       787787810
14         2023-06-15         33939       28024       832834733
15         2023-06-16         27268       22986       666181565
16         2023-06-17         11072        9164       241623746
17         2023-06-18          9584        7704       227903851
18         2023-06-19         32560       27151       817900332
19         2023-06-20         35581       29716       878959736
20         2023-06-21         31341       26044       771028759
21         2023-06-22         32090       26924       794670594
22         2023-06-23         25253       21308       610753955
23         2023-06-24         10371        8609       234221753
24         2023-06-25          8822        7100       210500533
25         2023-06-26         31086       26112       793722201
26         2023-06-27         33405       27946       841129924
27         2023-06-28         29144       24478       722759693
28         2023-06-29         30964       26072       773611764
29         2023-06-30         24128       20540       577824438

Logiquement, il y a très peu de différence entre le nombre de trajets et le nombre de passagers. Comme vu précédemment, 98% des réservations concernent un seul passager.